vllm/model_executor/layers/linear.py

```diff
@@ -29,19 +29,24 @@ class LinearMethodBase(ABC):
     """Base class for different (maybe quantized) linear methods."""
 
     @abstractmethod
-    def create_weights(self, input_size_per_partition: int,
+    def create_weights(self, layer: torch.nn.Module,
+                       input_size_per_partition: int,
                        output_size_per_partition: int, input_size: int,
-                       output_size: int,
-                       params_dtype: torch.dtype) -> Dict[str, Any]:
-        """Create weights for a linear layer."""
+                       output_size: int, params_dtype: torch.dtype,
+                       **extra_weight_attrs):
```

Comment:
Could this just be an optional `weight_loader`?

I see the value of enabling `kwargs` here for future extensibility, but I don't see a case that exists yet other than `weight_loader` so far

Perhaps making the argument explicit is better until we have a reason to allow kwargs

Reply:
Actually, making it a kwarg will be easier as we don't need extra handling to _not_ set the `weight_loader` if it's left unspecified.

Reply:
:)